nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/chronicle/ChronicleCompoundLogTailer.java

```diff
@@ -0,0 +1,166 @@
+        tailers.forEach(partition -> logPartitions.addAll(partition.assignments()));
+    }
+
+    @Override
+    public LogRecord<M> read(Duration timeout) throws InterruptedException {
+        LogRecord<M> ret = read();
+        if (ret != null) {
+            return ret;
+        }
+        final long timeoutMs = timeout.toMillis();
+        final long deadline = System.currentTimeMillis() + timeoutMs;
+        final long delay = Math.min(ChronicleLogTailer.POLL_INTERVAL_MS, timeoutMs);
+        while (ret == null && System.currentTimeMillis() < deadline) {
+            Thread.sleep(delay);
+            ret = read();
+        }
+        return ret;
+    }
+
+    protected LogRecord<M> read() {
+        if (size <= 0) {
+            return null;
+        }
+        // round robin on tailers
+        LogRecord<M> ret;
+        long end = counter + size;
+        do {
+            counter++;
+            int i = (int) counter % size;
+            ret = tailers.get(i).read();
+            if (ret != null) {
+                return ret;
+            }
+        } while (counter < end);
+        return null;
+    }
+
+    @Override
+    public LogOffset commit(LogPartition partition) {
+        for (LogTailer<M> tailer : tailers) {
+            if (tailer.assignments().contains(partition)) {
+                return tailer.commit(partition);
+            }
+        }
+        throw new IllegalArgumentException("No tailer matching: " + partition);
+    }
+
+    @Override
+    public void commit() {
+        tailers.forEach(LogTailer::commit);
+    }
+
+    @Override
+    public void toEnd() {
+        tailers.forEach(ChronicleLogTailer::toEnd);
+    }
+
+    @Override
+    public void toStart() {
+        tailers.forEach(ChronicleLogTailer::toStart);
+    }
+
+    @Override
+    public void toLastCommitted() {
+        tailers.forEach(ChronicleLogTailer::toLastCommitted);
+    }
+
+    @Override
+    public Collection<LogPartition> assignments() {
+        return logPartitions;
+    }
+
+    @Override
+    public String group() {
+        return group;
+    }
+
+    @Override
+    public boolean closed() {
+        return closed;
+    }
+
+    @Override
+    public void seek(LogOffset offset) {
+        for (LogTailer<M> tailer : tailers) {
+            if (tailer.assignments().contains(offset.partition())) {
+                tailer.seek(offset);
+                return;
+            }
+        }
+        // Should be an IllegalArgumentException but Kafka raise a state exception so do the same
+        throw new IllegalStateException("Cannot seek, tailer " + this + " has no assignment for partition: " + offset);
+    }
+
+    @Override
+    public void reset() {
+        tailers.forEach(ChronicleLogTailer::reset);
+    }
+
+    @Override
+    public void reset(LogPartition partition) {
+        ChronicleLogTailer<M> tailer = tailers.stream().filter(t -> t.assignments().contains(partition)).findFirst().orElse(null);
```

Comment:
Don't use `orElse(null)` and leverage on `Optional#isPresent` or `Optional#orElseThrown`